share/wake/lib/system/cp.wake

```diff
@@ -29,5 +29,21 @@ export def installAs dest file = match file.getPathResult
     | runJob
     | getJobOutput
 
-export def installIn dir file =
-  installAs "{dir}/{file.getPathName}" file
+# Copy a file from one directory subtree to another
+# installIn "/to" "from"  "from/subdir/file"  => "/to/subdir/file"
+# installIn "/to" "."     "from/subdir/file"  => "/to/from/subdir/file"
+# installIn "/to" "/from" "/from/subdir/file" => "/to/subdir/file"
+# installIn "/to" "/"     "/from/subdir/file" => "/to/from/subdir/file"
+# These fail:
+# installIn "/to" "from"  "subdir/file"
+# installIn "/to" "/from" "/subdir/file"
+# installIn "/to" "from"  "/from/subdir/file"
+# installIn "/to" "/from" "from/subdir/file"
+export def installIn toRoot fromRoot sourcePath =
+  require Path sourceStr = sourcePath
+  def rel = relative fromRoot sourceStr
+  def isAbs = matches `/.*`
+  def uneq a b = (a && !b) || (!a && b)
+  if matches `[.][.]/.*` rel || uneq fromRoot.isAbs sourceStr.isAbs
+  then makeBadPath (makeError "{sourceStr} is not contained in {fromRoot}")
```

Comment:
If we're going to change the API in this PR, could we also return a `Result`?

Reply:
Good point.

Reply:
Actually, I'd rather make this change when we change installAs. The grand elimination of BadPath will hopefully happen shortly, anyhow.